README.md

```diff
@@ -83,48 +253,71 @@ And then use the `encoded` value:
 make run-docker API_KEY="NGVPV2dZSUJBWWRNaUdIY3htS0g6eklKVGhGZzlUTzZ1YVlWeTU3VEZTQQ=="
 ```
 
-### Updating configuration values
+#### Local connector properties
 
-You can update the git branch, revision and project version in the `connectors.yml` file to your current values by
-running this command:
+Sensitive data, such as API keys, credentials, etc. could also be stored in the [configuration file](config/connectors.yml), provided that it's NOT under source control. Every connector reads this file on creation and is looking for the section called `<service_type>`, where `service_type` is the name that was provided in the corresponding property of the configuration file. You can look at it as a short identifier allowing you to specify which third-party service is behind the connector, or anything to help identify which connector you're running. Example:
 
+```yaml
+service_type: gitlab
+gitlab:
+  api_token: <your-api-key>
 ```
-make refresh_config
+
+This way, you're saying that the connector is for GitLab, and there you're providing the API key for GitLab. And it also means that the `GitLab::Connector` class will have a `@local_configuration` variable that will contain whatever is in the `gitlab` section of the configuration file. So inside the connector class, you can access the API key like this:
+
+```ruby
+@local_configuration[:api_token]
+```
+
+## Operating the connector
```

Comment:
I think we should be more explicit that connector clients can be deployed exclusively on self managed infrastructure. The end user should be aware from the beginning that they're required to manage the underlying infrastructure for the deployed connector client.